hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/HoodieReadClient.java

```diff
@@ -144,7 +145,15 @@ private void assertSqlContext() {
 
     // record locations might be same for multiple keys, so need a unique list
     Set<String> uniquePaths = new HashSet<>(paths);
-    Dataset<Row> originalDF = sqlContextOpt.get().read().parquet(uniquePaths.toArray(new String[uniquePaths.size()]));
+    Dataset<Row> originalDF = null;
+    if (uniquePaths.size() == 0) {
```

Comment:
merge with `paths.get(0).endsWith(HoodieFileFormat.PARQUET.getFileExtension()`?

Reply:
The `paths` here could be an empty List. And the unit test `TestHoodieReadClient`  expects an `AnalysisException` thrown when `paths` is an empty List. If we merge with `paths.get(0).endsWith(HoodieFileFormat.PARQUET.getFileExtension()`, we will have `ArrayIndexOutOfBoundsException` instead of  `AnalysisException` if `paths` is empty.

Reply:
here what i mean is `if (uniquePaths.size() == 0 || paths.get(0).endsWith(HoodieFileFormat.PARQUET.getFileExtension()) `.

Reply:
OK. Got it. Will update the PR.

Reply:
any update here?

Reply:
Updated. Please review. Thanks.